File: Python/Data Structures & Algorithms Module - Python/Week10/BST Attempt 2/BSTApp.py
"""
BSTApp.py
Billy Cussen
25/11/2020
"""

from BST import BST

myBST = BST()

print("Empty: "+str(myBST.isEmpty()))

try:
    print("Size: "+str(myBST.size()))
except Exception as e:
    print(e)
try:
    myBST.inOrder()
except Exception as e:
    print(e)
try:
    print("Min: "+str(myBST.min()))
except Exception as e:
    print(e)
try:
    print("Max: "+str(myBST.max()))
except Exception as e:
    print(e)
try:
    print("Contains 2: "+str(myBST.contains(2)))
except Exception as e:
    print(e)
try:
    print("Get Node: "+str(myBST.getNode(2)))
except Exception as e:
    print(e)
try:
    print("Get Parent: "+str(myBST.findParent(2)))
except Exception as e:
    print(e)

try:
    print("Remove: "+str(myBST.remove(2)))
except Exception as e:
    print(e)

myBST.insert(15)
myBST.insert(10)
myBST.insert(5)
myBST.insert(7)
myBST.insert(2)
myBST.insert(1)

myBST.insert(20)
myBST.insert(17)
myBST.insert(22)


print("\n\nEmpty: "+str(myBST.isEmpty()))

try:
    print("Size: "+str(myBST.size()))
except Exception as e:
    print(e)
try:
    myBST.inOrder()
except Exception as e:
    print(e)
try:
    print("Min: "+str(myBST.min()))
except Exception as e:
    print(e)
try:
    print("Max: "+str(myBST.max()))
except Exception as e:
    print(e)
try:
    print("Contains 2: "+str(myBST.contains(2)))
except Exception as e:
    print(e)
try:
    print("Contains 3000: "+str(myBST.contains(3000)))
except Exception as e:
    print(e)
try:
    print("Get Node: "+str(myBST.getNode(2)))
except Exception as e:
    print(e)
try:
    print("Get Parent: "+str(myBST.findParent(2)))
except Exception as e:
    print(e)
try:
    print("Get Parent: "+str(myBST.findParent(15)))
except Exception as e:
    print(e)

try:
    print("Remove: "+str(myBST.remove(2))+"\n")
except Exception as e:
    print(e)

try:
    myBST.inOrder()
except Exception as e:
    print(e)
try:
    print("Remove: "+str(myBST.remove(15000))+"\n")
except Exception as e:
    print(e)
try:
    print("Remove: "+str(myBST.remove(10)))
except Exception as e:
    print(e)

try:
    print("Remove: "+str(myBST.remove(20))+"\n")
except Exception as e:
    print(e)

try:
    myBST.inOrder()
except Exception as e:
    print(e)

File: Java/Data Structures & Algorithms Module - Java/Revision/BinarySearchTree2/BSTApp.java
/*
BSTApp.java
Billy Cusseb
25/01/2021
*/
public class BSTApp {
    
    public static void main(String [] args){

        BST<Integer> myBST = new BST<Integer>();

        myBST.inOrder();
        System.out.println("Max: "+myBST.max());
        System.out.println("Min: "+myBST.min());
        System.out.println("Size: "+myBST.size());
        System.out.println("Depth: "+myBST.depth());
        System.out.println("Contains 2: "+myBST.contains(2));
        System.out.println("Contains 18: "+myBST.contains(18));
        System.out.println("Contains 35: "+myBST.contains(35));
        System.out.println("Contains 17: "+myBST.contains(17));
        myBST.insert(20);
        myBST.insert(15);
        myBST.insert(17);
        myBST.insert(10);
        myBST.insert(30);
        myBST.insert(35);
        myBST.insert(40);
        myBST.inOrder();
        System.out.println("Max: "+myBST.max());
        System.out.println("Min: "+myBST.min());
        System.out.println("Size: "+myBST.size());
        System.out.println("Depth: "+myBST.depth());
        System.out.println("Contains 2: "+myBST.contains(2));
        System.out.println("Contains 18: "+myBST.contains(18));
        System.out.println("Contains 35: "+myBST.contains(35));
        System.out.println("Contains 17: "+myBST.contains(17));
    }

}
